src/ConformalTracking.cc

```diff
@@ -1659,6 +1676,14 @@ void ConformalTracking::extendHighPT(UniqueKDTracks& conformalTracks, SharedKDCl
     streamlog_out(DEBUG7) << "- track finishes with size " << conformalTracks[currentTrack]->m_clusters.size() << std::endl;
   }
 
+  // Assign kalman fit direction based on the parameters
+  bool kalmanFitBackward = parameters._kalmanFitBackward;  // true if backward is set
+
+  for (auto& ct : conformalTracks) {
```

Comment:
This loops over all the conformalTracks created up to this point. Not just those that are created in the step that calls this function, because the conformalTracks collection is always the same.

Reply:
yes, and this is fine. The tracks built in "build" and later on extended in "extend" will have an ultimate fit direction set according to the parameter parsed to the "extend" function.

Reply:
If you have step1 extend + fitbackward and step2 extend  + fitForward, all the tracks created in step1 will now also have fitForward

Reply:
there's only one extend per build, and typically build and extend would have the same fit direction set (I don't see the reason for otherwise settings)

Reply:
If you have step1 build + extend + fitbackward and step2 build + extend + fitForward, all the tracks created in step1 will now also have fitForward